### PR TITLE
Batch account lookups for sparse ledger creation

### PR DESCRIPTION
**Pull request by mrmr1993**
Checklist:

- [ ] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

### CODE REVIEW

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-nightly-me

**Comment by mrmr1993** [comment]
!ci-build-me

File: src/lib/merkle_mask/masking_merkle_tree.ml [hunk 1]
```diff
@@ -185,15 +185,44 @@ module Make (Inputs : Inputs_intf.S) = struct
...
     let get_batch t locations =
       assert_is_attached t ;
-      let found_accounts, leftover_locations =
-        List.partition_map locations ~f:(fun location ->
-            match self_find_account t location with
-            | Some account ->
-                Either.first (location, Some account)
-            | None ->
-                Either.second location )
+      let locations_with_locations_rev, leftover_locations_rev =
```
Comment (georgeee) [inline src/lib/merkle_mask/masking_merkle_tree.ml:188]: I suggest using the [version](https://github.com/MinaProtocol/mina/commit/8a5022369d5ae850cd882c3c681bd129eb387822)

**Comment by nholland94** [comment]
!ci-build-me

File: src/lib/mina_ledger/sparse_ledger.ml [hunk 2]
```diff
@@ -9,21 +9,32 @@ let of_ledger_root ledger =
...
-            , add_path sl
-                (Ledger.merkle_path ledger loc)
-                key
-                ( Ledger.get ledger loc
-                |> Option.value_exn ?here:None ?error:None ?message:None ) )
-        | None ->
-            let path, acct = Ledger.create_empty_exn ledger key in
-            (key :: new_keys, add_path sl path key acct) )
-      ~init:([], of_ledger_root ledger)
+    let rec go (new_keys, sl) locations accounts =
```
Comment (georgeee) [inline src/lib/mina_ledger/sparse_ledger.ml]: Could be better structured, but will be amended in a PR up the PR chain

**Review by georgeee — APPROVED** [review]

**Review by ghost-not-in-the-shell — APPROVED** [review]

**Review by nholland94 — APPROVED** [review]

**Comment by nholland94** [comment]
!ci-build-me

**Comment by deepthiskumar** [comment]
nightly pipeline passes https://buildkite.com/o-1-labs-2/mina-end-to-end-nightlies/builds/877#_

**Comment by deepthiskumar** [comment]
!approved-for-mainnet

**Pull request merged**